lerobot/common/datasets/push_dataset_to_hub/gpr_krec_format.py

```diff
@@ -0,0 +1,321 @@
+"""
+This script loads a GPR dataset from KREC files and converts it to lerobot dataset format.
+
+Example Usage:
+    python lerobot/common/datasets/push_dataset_to_hub/gpt_krec_format.py --raw_dir /path/to/krec/files
+    
+    python lerobot/common/datasets/push_dataset_to_hub/gpt_krec_format.py --raw_dir /home/kasm-user/ali_repos/kmodel/data/datasets/krec_data/dec_3__11_10am_og_krecs_edited/2024-12-03_17-47-30/
+"""
+
+import argparse
+from pathlib import Path
+from pprint import pprint
+
+import os
+from datetime import datetime
+
+import h5py
+import numpy as np
+import torch
+from datasets import Dataset, Features, Sequence, Value
+from tqdm import tqdm
+
+import krec
+from scipy.spatial.transform import Rotation as R
+
+
+from lerobot.common.datasets.lerobot_dataset import CODEBASE_VERSION
+from lerobot.common.datasets.push_dataset_to_hub.utils import (
+    calculate_episode_data_index,
+    concatenate_episodes,
+)
+from lerobot.common.datasets.utils import hf_transform_to_torch
+
+def get_krec_file_type(file_path: str) -> str:
+    """Determine if the file is a direct KREC file or MKV-embedded KREC.
+
+    Returns:
+        'krec' for .krec files
+        'mkv' for .krec.mkv files
+        raises RuntimeError for invalid extensions
+    """
+    if file_path.endswith(".krec"):
+        return "krec"
+    elif file_path.endswith(".krec.mkv"):
+        return "mkv"
+    else:
+        error_msg = (
+            f"Invalid file extension. Expected '.krec' or '.krec.mkv', got: {file_path}"
+        )
+        raise RuntimeError(error_msg)
+
+
+def load_krec_direct(krec_file_path: str) -> krec.KRec:
+    """Load a KREC file directly."""
+    return krec.KRec.load(krec_file_path)
+
+
+def load_krec_from_mkv(mkv_file_path: str) -> krec.KRec:
+    """Load a KREC file from an MKV file into a manually created temp directory."""
+
+    if not os.path.exists(mkv_file_path):
+        raise FileNotFoundError(f"File not found: {mkv_file_path}")
+
+    # Create a parent temp directory if it doesn't exist
+    parent_temp_dir = os.path.join(os.path.dirname(mkv_file_path), "temp")
+    os.makedirs(parent_temp_dir, exist_ok=True)
+
+    # Create timestamped subdirectory inside parent temp directory
+    timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    temp_dir = os.path.join(parent_temp_dir, f"temp_{timestamp}")
+    os.makedirs(temp_dir, exist_ok=True)
+
+    base_name = os.path.basename(mkv_file_path).split(".krec.mkv")[0]
+    krec_file_path = os.path.join(temp_dir, f"{base_name}_from_mkv.krec")
+
+    # Extract and load from temp directory
+    krec.extract_from_video(mkv_file_path, krec_file_path)
+    return krec.KRec.load(krec_file_path)
+
+
+def load_krec(file_path: str) -> krec.KRec:
+    """Smart loader that handles both direct KREC and MKV-embedded KREC files."""
+    file_type = get_krec_file_type(file_path)
+
+    if file_type == "krec":
+        return load_krec_direct(file_path)
```

Comment:
```suggestion
        return krec.KRec.load(krec_file_path)
```
simplicity always wins, no langchain patterns